tempesta_fw/http.c

```diff
@@ -2588,20 +2598,14 @@ tfw_http_resp_cache(TfwHttpMsg *hmresp)
 	}
 
 	/*
-	 * This hook isn't in tfw_http_resp_fwd() because it isn't needed
-	 * to count responses from a cache.
-	 * FSM needs TfwHttpReq to record data. It needs TfwHttpResp as well.
-	 * It will be added to the request later in tfw_http_resp_fwd(), but it's
-	 * needed now. Save previous (NULL) value for the sake of safety
+	 * This hook isn't in tfw_http_resp_fwd() because responses from the
+	 * cache shouldn't be accounted.
 	 */
-	state = &hmresp->conn->state;
-	prev_state_obj = state->obj;
-	state->obj = req;
-	prev_resp = req->resp;
-	req->resp = hmresp;
-	tfw_gfsm_move(state, TFW_HTTP_FSM_RESP_MSG_FWD, NULL, 0);
-	state->obj = prev_state_obj;
-	req->resp = prev_resp;
+	data.skb = NULL;
+	data.off = 0;
+	data.req = (TfwMsg *)req;
+	data.resp = (TfwMsg *)hmresp;
+	tfw_gfsm_move(&hmresp->conn->state, TFW_HTTP_FSM_RESP_MSG_FWD, &data);
```

Comment:
The hook was added in #717, recently closed #520 required the same for cached responses, according to [Wiki](https://github.com/tempesta-tech/tempesta/wiki/DDoS-mitigation#web-cache-bypass) the feature is required. Sorry but I got confused, is it required to support the #520? Now it's not supported.
If i's required, then either the same gfsm hook must be called in `tfw_http_req_cache_service()` or this hook must be moved to  `tfw_http_resp_fwd()`

Reply:
The idea was that if we limit number of negative server response for a client, then it's enough for both of the issues, #520 and #717. The only difference between them is that in #717'th case there is typically the same URI in all client requests while in #520'th URIs are different. There is no need to care about Web cache filling if a client hitted the limit - just let a garbage collector (subject for TDBv2) collect all negative and useless cache entries.